src/zenml/models/model_base_model.py

```diff
@@ -64,18 +70,91 @@ class ModelBaseModel(BaseModel):
+    version_number: points model context to a specific version number.
+    version_description: The description of the model version.
     create_new_model_version: Whether to create a new model version during execution
     save_models_to_registry: Whether to save all ModelArtifacts to Model Registry,
         if available in active stack.
     delete_new_version_on_failure: Whether to delete failed runs with new versions for later recovery from it.
     """
 
-    version: Optional[Union[ModelStages, str]] = Field(
+    version_name: Optional[Union[ModelStages, str]] = Field(
         default=None,
         description="Model version or stage is optional and points model context to a specific version/stage, "
         "if skipped and `create_new_model_version` is False - latest model version will be used.",
     )
+    version_number: Optional[int] = Field(
+        default=None,
+        description="Model version number is optional and points model context to a specific version number, "
+        "this works only for reading from model version and not suitable with `create_new_model_version`.",
+    )
+    version_description: Optional[str]
     create_new_model_version: bool = False
     save_models_to_registry: bool = True
     delete_new_version_on_failure: bool = True
+
+    class Config:
+        """Config class."""
+
+        smart_union = True
+
+    @root_validator
+    def _root_validator(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate all in one.
+
+        Args:
+            values: Dict of values.
+
+        Returns:
+            Dict of validated values.
+
+        Raises:
+            ValueError: If validation failed on one of the checks.
+        """
+        create_new_model_version = values.get(
+            "create_new_model_version", False
+        )
+        delete_new_version_on_failure = values.get(
+            "delete_new_version_on_failure", True
+        )
+        if not delete_new_version_on_failure and not create_new_model_version:
+            logger.warning(
+                "Using `delete_new_version_on_failure=False` and `create_new_model_version=False` has no effect."
+                "Setting `delete_new_version_on_failure` to `True`."
+            )
+            values["delete_new_version_on_failure"] = True
+
+        version_number = values.get("version_number", None)
+        version_name = values.get("version_name", None)
+
+        if version_number and create_new_model_version:
+            raise ValueError(
+                "`version_number` cannot be used with `create_new_model_version`."
+            )
+
+        if version_number is not None and version_name is not None:
+            logger.warning(
+                "`version_number` has higher priority then `version_name`."
+                "Setting `version_name` to `None`."
+            )
+            version_name = None
+            values["version_name"] = None
```

Comment:
Same here - feels a bit clunky from a UX perspective

Reply:
I simplified it 👍🏼